### PR TITLE
Feat: Style Tiles component

### PR DESCRIPTION
**Pull request by eduardoformiga**
## What's the purpose of this pull request?
- [x] Style the `Tiles` component. 
- [x] Handle the 2 tiles case, so it was necessary to change the component a little.
- [x] Add the component to the `pattern-library`.
- [x] Integrate products in the `Homepage`.
- [x] Integrate `ProductCard` + `Tiles`.
- [x] Integrate `Tiles` in the `Homepage`. 

follow up #141 

* on hold until we can test with internal content, like a `ProductCard`.

## How it works? 

|2 Tile|3 Tile|4 Tile|
|-|-|-|
|tiles 2 items|tiles3|tiles4|
|tiles2|tiles3|tiles4|

## How to test it?
You can see in the root homepage, and also the structure in the `/pattern-library` page

https://preview-175--base.preview.vtex.app/pattern-library/

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -19,6 +23,23 @@ function Page(props: Props) {
 
   const title = site?.siteMetadata?.title ?? ''
   const siteUrl = `https://${host}${pathname}`
+  const perPage = 10
+  const productList = useProductsQuery({
+    first: perPage,
+    after: (perPage * 1).toString(),
+    sort: 'score_desc',
+    term: '',
+    selectedFacets: [{ key: 'category-1', value: 'office' }],
```
Comment (igorbrasileiro) [inline src/pages/index.tsx:32]: Are these selected facets would be intended to be hardcoded?

**Review by igorbrasileiro — APPROVED** [review]

File: src/pages/index.tsx (hunk 1 above)
Comment (eduardoformiga) [reply src/pages/index.tsx:32]: At this moment in an alpha version, this query is to get a real example about how to get products on the homepage.
As we define the products to be shown we can change the query.

File: src/pages/index.tsx [hunk 2]
```diff
@@ -19,6 +23,23 @@ function Page(props: Props) {
...
+  })
+
+  const products = useMemo(
+    () => productList?.edges.map((edge) => edge.node),
+    [productList]
+  )
+
+  if (products == null) {
+    return null
```
Comment (filipewl) [inline src/pages/index.tsx:41]: Will this make the whole component not render when no products are returned? It seems too drastic if that's the case. Can't we scope this to where `products` is used (in `ProductTiles` below) so it only affects that area of the page? Something like `products && <ProductTiles products={products} />`?

**Review by filipewl — APPROVED** [review]
Nice!

Screen Shot 2021-12-30 at 16 08 41

File: src/pages/index.tsx (hunk 2 above)
Comment (eduardoformiga) [reply src/pages/index.tsx:41]: You are correct! It looks so drastic hahaha. I'm gonna change it. Thanks!

**Review by renatamottam — APPROVED** [review]
LATM! (Looks Awesome To Me)

File: src/components/ui/Tiles/tiles.scss [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+[data-store-tiles="expanded-first"] > [data-store-tile]:first-child,
+[data-store-tiles="expanded-first-two"] > [data-store-tile]:first-child,
+[data-store-tiles="expanded-first-two"] > [data-store-tile]:nth-child(2) {
+  grid-column: span 2;
+  min-width: 12rem;
+  height: fit-content;
+
+  [data-store-card-image] {
+    height: calc((100vw - 2rem) * .65);
```
Comment (hellofanny) [inline src/components/ui/Tiles/tiles.scss:89]: 👀 How does it work? Is there any value that we can use as variable?

Comment (renatamottam) [reply src/components/ui/Tiles/tiles.scss:89]: hahahahah it's a fix for now, but we should start using aspect ratio on those images.

**Review by hellofanny — APPROVED** [review]
LGTM! (Looks Gorgeous To Me) 🤩

File: src/components/ui/Tiles/tiles.scss (hunk 3 above)
Comment (eduardoformiga) [reply src/components/ui/Tiles/tiles.scss:89]: It's a magic made by @renatamottam 🪄  hahahah

**Comment by eduardoformiga** [comment]
Unfortunately, the CI LTTM (Looks Terrify To Me) hahahaha

**Pull request merged**